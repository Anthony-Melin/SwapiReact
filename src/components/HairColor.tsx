import LabelValueFactory from "./__components__/LabelValueFactory.tsx";

export default LabelValueFactory('haircolor', 'Hair color')
