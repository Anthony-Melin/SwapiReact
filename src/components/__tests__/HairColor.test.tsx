import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import HairColor from '../HairColor.tsx'

describe('HairColor', () => {
    beforeEach(() => {
        render(<HairColor value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('haircolor')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('haircolor')).toHaveTextContent(
            'Hair color: VALUE'
        )
    })
})
