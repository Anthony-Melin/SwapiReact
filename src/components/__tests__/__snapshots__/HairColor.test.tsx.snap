// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HairColor > should match snapshot 1`] = `
<p
  data-testid="haircolor"
>
  Hair color
  : 
  VALUE
</p>
`;
